lib/p2p/NodeList.ts

```diff
@@ -87,6 +103,28 @@ class NodeList {
     return false;
   }
 
+  /**
+   * Add a reputation event to the nodes history
+   * @return true if the node was banned, false otherwise
+   */
+  public addReputationEvent = async (nodePubKey: string, type: ReputationEventType) => {
+    const node = this.nodes.get(nodePubKey);
+    if (node) {
+      node.reputationScore += ReputationEventWeight[type];
+
+      if (this.nodeIsBanned(node)) {
+        this.emit('node.ban', nodePubKey, node.reputationEvents);
+      }
+
+      node.reputationEvents = [...node.reputationEvents, { type, date: ms() }];
```

Comment:
I find this syntax a little unusual for adding an element to an array vs just using `push()`. And out of curiosity I ran some benchmarks and using `push` is also much much more performant, I'm guessing that this way is cloning the `reputationEvents` array each time. http://jsben.ch/JpVcR